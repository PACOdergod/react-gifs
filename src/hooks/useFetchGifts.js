import { useState } from "react"
import { useEffect } from "react/cjs/react.development";
import { getGifs } from "../helpers/getGifs";


export const useFetchGifts = ( categoria )=>{
    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect( ()=>{

        getGifs( categoria ).then( imgs =>{
            setTimeout(() => {
                setState({
                    data: imgs,
                    loading: false
                }) 
            },0);
        })

    }, [ categoria ])


    return state
}