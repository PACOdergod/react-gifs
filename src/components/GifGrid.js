
import React from 'react'
import PropTypes from 'prop-types'
import { GifItem } from './GifItem'
import { useFetchGifts } from '../hooks/useFetchGifts'

const GifGrid = ({ categoria }) => {

    const { data:images, loading } = useFetchGifts( categoria )
    

    return (
        <>
            <h3>{ categoria }</h3>
            { loading && <p className="animate__animated animate__flash" >Loading</p> }
            <div className="card-grid">
                {
                    images.map( img => {
                        return <GifItem { ...img } key={img.id}/>
                    })
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    categoria : PropTypes.string.isRequired
}

export default GifGrid
