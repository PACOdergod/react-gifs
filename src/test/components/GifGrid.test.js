import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'

import GifGrid from '../../components/GifGrid';
import { useFetchGifts } from '../../hooks/useFetchGifts';
jest.mock('../../hooks/useFetchGifts')


describe('Pruebas de <GifGrid />', () => {
    
    
    test('debe dibujarse correctamente', () => {

        useFetchGifts.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow( <GifGrid categoria="perros"/>)
        expect( wrapper ).toMatchSnapshot()
    })

    test('debe de mostrar items cuando se cargan imagenes useFetchGifts', () => {

        const gifs = [{
            id: 'qwesda',
            url: 'https://google.com',
            title: 'titulo'
        }]

        useFetchGifts.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        
        const wrapper = shallow( <GifGrid categoria="perros"/>)

        expect( wrapper.find('p').exists() ).toBe( false )
        expect( wrapper.find('GifItem').length ).toBe( gifs.length )

    })
    
    
})
