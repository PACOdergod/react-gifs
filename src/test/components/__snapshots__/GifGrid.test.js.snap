// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGrid /> debe dibujarse correctamente 1`] = `
<Fragment>
  <h3>
    perros
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
